lib/src/scalafmt/java/com/diffplug/spotless/glue/scalafmt/ScalafmtFormatterFunc.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 DiffPlug
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless.glue.scalafmt;
+
+import java.io.File;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+
+import org.scalafmt.Scalafmt;
+import org.scalafmt.config.ScalafmtConfig;
+import org.scalafmt.config.ScalafmtConfig$;
+
+import com.diffplug.spotless.FileSignature;
+import com.diffplug.spotless.FormatterFunc;
+
+import scala.collection.immutable.Set$;
+
+public class ScalafmtFormatterFunc implements FormatterFunc {
+	private final FileSignature configSignature;
+
+	public ScalafmtFormatterFunc(FileSignature configSignature) {
+		this.configSignature = configSignature;
+	}
+
+	@Override
+	public String apply(String input) throws Exception {
+		ScalafmtConfig config;
+		if (configSignature.files().isEmpty()) {
+			// Note that reflection is used here only because Scalafmt has a method called
+			// default which happens to be a reserved Java keyword. The only way to call
+			// such methods is by reflection, see
+			// https://vlkan.com/blog/post/2015/11/20/scala-method-with-java-reserved-keyword/
+			Method method = ScalafmtConfig$.MODULE$.getClass().getDeclaredMethod("default");
+			config = (ScalafmtConfig) method.invoke(ScalafmtConfig$.MODULE$);
+		} else {
+			File file = configSignature.getOnlyFile();
+			String configStr = new String(Files.readAllBytes(file.toPath()), StandardCharsets.UTF_8);
+			config = Scalafmt.parseHoconConfig(configStr).get();
+		}
+		return Scalafmt.format(input, config, Set$.MODULE$.empty()).get();
```

Comment:
As a quality of life improvement, its possible to unroll the inner `NoSuchElementException` that is thrown by `.get()` so that the real exception is thrown (such as having an invalid configuration).

Do you want me to do this?

Reply:
That would be great!

Reply:
So I just had a look at the code behind `.get()` and its technically not an exception. More specifically its actually the underlying `Configuration` is an `Either` with its `Right` value being a `NotOk` which **can** be an exception but doesn't have to be with the `.get` method just putting a `.toString` representation in a `NoSuchElementException`.

Due to this I think its better off just leaving it as is.